go/obscuronode/enclave/enclave.go

```diff
@@ -47,11 +47,20 @@ type Enclave interface {
 	RoundWinner(parent common3.L2RootHash) (common2.ExtRollup, bool)
 	Stop()
 
-	// TestPeekHead - only available for testing purposes
-	TestPeekHead() BlockState
+	// L1Height returns the current L1 height
+	L1Height() int
```

Comment:
I'd say to move these L1Height and Hash methods out of the enclave. This information is in no way secret or encrypted.
You can move them to the Obscuro "node" backed by some datastructure populated on the notification from the L1 node

Reply:
I was planning on deferring that to another PR, thinking it would take me longer than it did ! 👍